libraries/Filter/NotchFilter.cpp

```diff
@@ -44,26 +44,36 @@ void NotchFilter<T>::init(float sample_freq_hz, float center_freq_hz, float band
     if ((center_freq_hz > 0.5 * bandwidth_hz) && (center_freq_hz < 0.5 * sample_freq_hz)) {
         float A, Q;
         calculate_A_and_Q(center_freq_hz, bandwidth_hz, attenuation_dB, A, Q);
-        init_with_A_and_Q(sample_freq_hz, center_freq_hz, A, Q);
+        init_with_A_and_Q(sample_freq_hz, center_freq_hz, A, Q, 1.0f);
     } else {
         initialised = false;
     }
 }
 
 template <class T>
-void NotchFilter<T>::init_with_A_and_Q(float sample_freq_hz, float center_freq_hz, float A, float Q)
+void NotchFilter<T>::init_with_A_and_Q(float sample_freq_hz, float center_freq_hz, float A, float Q, float max_slew)
 {
-    if ((center_freq_hz > 0.0) && (center_freq_hz < 0.5 * sample_freq_hz) && (Q > 0.0)) {
-        float omega = 2.0 * M_PI * center_freq_hz / sample_freq_hz;
+    float new_center_freq = center_freq_hz;
+
+    // constrain the new center frequency by a percentage of the old frequency
+    if (initialised && !need_reset && !is_zero(max_slew)) {
+        new_center_freq = constrain_float(new_center_freq, _center_freq_hz * (1.0 - max_slew),
+                                          _center_freq_hz * (1.0 + max_slew));
```

Comment:
This should be:
if is_positive(max_slew) {
     new_center_freq = constrain_float(new_center_freq, _center_freq_hz * (1.0 - max_slew),  _center_freq_hz * 1/(1.0 - max_slew));
}

Reply:
Fixed